filepathfilter/filepathfilter_test.go

```diff
@@ -55,34 +50,37 @@ func TestPatternMatch(t *testing.T) {
 	assertPatternMatch(t, "sub",
 		"sub/",
 		"sub",
-		"sub/filename.txt",
 		"top/sub/",
 		"top/sub",
+	)
+	refutePatternMatch(t, "sub",
+		"sub/filename.txt",
 		"top/sub/filename.txt",
 	)
 
-	assertPatternMatch(t, "sub/", "sub/filename.txt", "top/sub/filename.txt")
-	assertPatternMatch(t, "/sub", "sub/", "sub", "sub/filename.txt")
-	assertPatternMatch(t, "/sub/", "sub/filename.txt")
+	refutePatternMatch(t, "sub/", "sub/filename.txt", "top/sub/filename.txt")
+	assertPatternMatch(t, "/sub", "sub")
+	refutePatternMatch(t, "/sub", "sub/filename.txt")
+	refutePatternMatch(t, "/sub/", "sub/filename.txt")
```

Comment:
I've adopted this and the other one.  I think I may need to make some adjustments because I don't believe the trailing slash matches on Windows in the same way due to the difference in path separators.